src/nncf/experimental/common/tensor_statistics/collectors.py

```diff
@@ -555,25 +557,19 @@ def __hash__(self) -> int:
 
 
 class NoopAggregator(AggregatorBase):
-    def __init__(self, num_samples: Optional[int]):
+    def __init__(self, num_samples: Optional[int], return_first: bool = False):
+        if return_first and num_samples is not None and num_samples > 1:
+            msg = "NoopAggregator with return_first=True should not have num_samples > 1"
+            raise nncf.InternalError(msg)
+        num_samples = 1 if num_samples is None and return_first else num_samples
```

Comment:
Micro simplification
```suggestion
        num_samples = 1 if return_first else num_samples
```